app/src/main/java/com/denytheflowerpot/scrunch/services/FoldActionSignalingService.kt

```diff
@@ -3,88 +3,45 @@ package com.denytheflowerpot.scrunch.services
-                    lines.forEach { line ->
-                        if (!this.isActive) {
-                            throw DummyStopServiceException()
-                        }
-
-                        val folded = foldDetectionStrategy.processLogcatTrace(line, currentFoldState)
-                        if (folded != null) {
-                            Log.d("Scrunch", "Fold status is $folded")
-                            if (currentFoldState == null) {
-                                currentFoldState = folded
-                            } else {
-                                if (folded != currentFoldState) {
-                                    val soundPlaybackManager =
-                                        ScrunchApplication.instance.soundPlaybackManager
-                                    if (folded) soundPlaybackManager.playFoldSound() else soundPlaybackManager.playUnfoldSound()
-                                    currentFoldState = folded
-                                }
-                            }
-                        } else {
-                            Log.d("Scrunch", "Invalid line: $line")
-                        }
-                    }
-                } catch (e: Exception) {
-                    if (e !is DummyStopServiceException) {
-                        Log.d("Scrunch", "Error: $e")
-                    }
-                }
-            }
-    }
+    private var currentFoldState: Int? = null
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
         if (intent != null && intent.action == stopServiceAction) {
             stopSelf()
             return START_NOT_STICKY
         }
 
-        if (logcatJob == null) {
-            val strategy = FoldDetectionStrategy.instanceForThisDevice
-            return if (strategy != null) {
-                foldDetectionStrategy = strategy
-                scope = CoroutineScope(Job() + Dispatchers.IO)
-                logcatJob = scope.launch { announceFoldAction() }
-                startForeground(
-                    notificationId,
-                    ScrunchApplication.instance.notificationManager.generateNotification(
-                        stopServiceAction
-                    )
-                )
-                START_STICKY
-            } else {
-                Log.d("Scrunch", "Could not detect model")
-                START_NOT_STICKY
+        return START_STICKY
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        startForeground(
+            notificationId,
+            ScrunchApplication.instance.notificationManager.generateNotification(
+                stopServiceAction
+            )
+        )
+        FoldDetectionStrategy.instanceForThisDevice?.create(this) { state ->
+            if (state == DEVICE_STATE_HALF_OPEN) {
+                //Currently ignoring this state.
+                return@create
+            }
+
+            if (state != currentFoldState) {
```

Comment:
Nitpick: since this doesn't check whether `currentFoldState` has been initialized to any value yet, upon first starting up Scrunch it will play the sound for the current state of the device - e.g., if i open Scrunch and my device is folded it will play the fold sound. IMO if `currentFoldState` is null, it is better to assign it to the new value without playing a sound.